core/src/main/java/org/pentaho/di/connections/ConnectionManager.java

```diff
@@ -291,6 +293,26 @@ public <T extends ConnectionDetails> boolean test( T connectionDetails ) throws
     return connectionProvider.test( connectionDetails );
   }
 
+  /**
+   * Tests if a given VFS connection is valid, optionally, with certain testing options.
+   *
+   * @param connectionDetails The VFS connection.
+   * @param options The testing options, or {@code null}.
+   * @return {@code true} if the connection is valid; {@code false}, otherwise.
+   */
+  public <T extends VFSConnectionDetails> boolean test( T connectionDetails, VFSConnectionTestOptions options )
+          throws KettleException {
+    if ( Objects.isNull( connectionDetails ) ) {
```

Comment:
Instead of:
```java
if ( Objects.isNull( connectionDetails ) ) {
 ```

Why not, simply?
```java
if ( connectionDetails == null ) {
```

The `Objects.isNull` method exists to be used in functional programming contexts.

Reply:
addressed the change.